cdci_data_analysis/analysis/instrument.py

```diff
@@ -253,6 +255,56 @@ def parse_inputs_files(self,
 
             raise RequestNotUnderstood(error_message)
 
+    def get_status_details(self,
+                           par_dic,
+                           config=None,
+                           logger=None,
+                           sentry_client=None):
+        if logger is None:
+            logger = self.get_logger()
+
+        status_details = None
+
+        # TODO put this in a dedicated function, perhaps within the oda_api
+        # adaptation for oda_api, like it happens in oda_api set_api_code function
+        updated_par_dic = par_dic.copy()
+        updated_par_dic['product'] = updated_par_dic['product_type']
+        updated_par_dic['product_type'] = updated_par_dic['query_type']
+        updated_par_dic.pop('query_type')
+
+        logger.info(f"getting products for a more in-depth analysis for the results within run_call_back with args {updated_par_dic}")
+        disp = DispatcherAPI(url=config.dispatcher_callback_url_base, instrument='mock')
+        try:
+            disp.get_product(**updated_par_dic)
+        except (DispatcherException,
+                DispatcherNotAvailable,
+                UnexpectedDispatcherStatusCode,
+                RequestNotUnderstoodOdaApi) as de:
+            logger.info('A dispatcher-related exception has been returned from the oda_api when retrieving information '
+                        'from a completed job, this is probably related to an empty product')
+            if sentry_client is not None:
+                sentry_client.capture('raven.events.Message',
+                                      message=(f'Dispatcher-related exception detected when retrieving additional '
+                                               f'information from a completed job '
+                                               f'{de}'))
+        except ConnectionError as ce:
+            logger.info(f'A connection error exception has been detected when retrieving additional information from a completed job: {ce}')
+            if sentry_client is not None:
+                sentry_client.capture('raven.events.Message',
+                                      message=(f'ConnectionError detected when retrieving additional '
+                                               f'information from a completed job '
+                                               f'{ce}'))
+        except RemoteException as re:
+            logger.info(f'A connection error exception has been detected when retrieving additional information from a completed job: {re}')
```

Comment:
this is really not too useful for the user. See the example instead

Reply:
another problem is that `RemoteException` might occur with a correctly completed job with empty result

Reply:
> > another problem is that `RemoteException` might occur with a correctly completed job with empty result
> 
> And how can this be detected? By inspecting the content of the exception?

yes

Reply:
The inspection I perform now is based on the output returned by the `oda_api`.

In particular, searching for these two cases:
- https://github.com/oda-hub/oda_api/blob/master/oda_api/api.py#L158-L190 (probably connection error)
- https://github.com/oda-hub/oda_api/blob/master/oda_api/api.py#L813-L832 (probably empty results)